okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -493,6 +493,14 @@ private void onGoAway(int lastKnownStreamId, Status status) {
     boolean notifyShutdown;
     synchronized (lock) {
       notifyShutdown = !goAway;
+    }
+    if (notifyShutdown) {
+      // We call transportShutdown() here, because once the goAway becomes true, other thread may
+      // call stopIfNecessary() and cause listener.transportTerminated() be called.
```

Comment:
`ClientTransport.Listener` is not thread-safe, notifications to it must be [properly synchronized externally](https://github.com/grpc/grpc-java/blob/a508c1d4f5024c2a5fd6d103edcab43542293ae7/core/src/main/java/io/grpc/internal/ClientTransport.java#L102). If more than one thread is calling the listener, is there any synchronization between them?